test/lazy/test_ts_opinfo.py

```diff
@@ -0,0 +1,161 @@
+
+        torch.testing.assert_close(weight_copy_grad.cpu(), weight_grad.cpu())
+        torch.testing.assert_close(inp_copy_grad.cpu(), inp_grad.cpu())
+
+class TestLazyOpInfo(TestCase):
+
+    @ops([op for op in op_db if op.name in LAZY_OPS_LIST and op.name not in SKIP_RUNTIME_ERROR_LIST], allowed_dtypes=(torch.float,))
+    def test_dispatched_to_lazy(self, device, dtype, op):
+
+        def get_name(op):
+            l = [op.name]
+            if op.variant_test_name != '':
+                l.append(op.variant_test_name)
+            return '.'.join(l)
+
+        global FALLBACK_LIST
+        samples = op.sample_inputs("lazy", dtype, requires_grad=False)
+        sample = list(samples)[0]
+        args = [sample.input] + list(sample.args)
+        kwargs = sample.kwargs
+        torch._lazy.mark_step()
+        torch._lazy.wait_device_ops()
+        torch._lazy.metrics.reset()
+
+        r = op(*args, **kwargs)
+        torch._lazy.mark_step()
+        torch._lazy.wait_device_ops()
+        prefix = "aten" if op.name in FALLBACK_LIST else "lazy"
+        found = f"{prefix}::{op.name}" in remove_suffixes(torch._lazy.metrics.counter_names())
+        # check aliases
+        if not found:
+            for alias in op.aliases:
+                alias_found = f"{prefix}::{alias.name}" in remove_suffixes(torch._lazy.metrics.counter_names())
+                found = found or alias_found
+                if found:
+                    break
+        self.assertTrue(found)
+
+
+    @ops([op for op in op_db if op.name in LAZY_OPS_LIST and op.name not in SKIP_RUNTIME_ERROR_LIST | SKIP_INCORRECT_RESULTS_LIST], allowed_dtypes=(torch.float,))  # noqa: B950
+    def test_correctness(self, device, dtype, op):
+
+
+        test_device = get_test_device()
+
+        def clone_to_device(input, dev):
+            if isinstance(input, torch.Tensor):
+                return input.detach().clone().to(device=dev)
+            if isinstance(input, Sequence) and not isinstance(input, str):
+                return tuple(map(functools.partial(clone_to_device, dev=dev), input))
+            return input
+
+        def assert_allclose_rec(t):
+            a, b = t
+            self.assertEqual(type(a), type(b))
+            if isinstance(a, torch.Tensor):
+                self.assertTrue(torch.allclose(clone_to_device(a, test_device), b, atol=1e-4))
+
+            if isinstance(a, Sequence):
+                map(assert_allclose_rec, zip(a, b))
+
+        samples = op.sample_inputs("lazy", dtype, requires_grad=False)
+        for sample in samples:
+            args = [sample.input] + list(sample.args)
+            kwargs = sample.kwargs
+            copy_args = clone_to_device(args, test_device)
+
+            r_exp = op(*copy_args, **kwargs)
+            r_actual = op(*args, **kwargs)
+
+            assert_allclose_rec((r_actual, r_exp))
+
+# TODO: after we move to master, add Lazy as a new Device here:
```

Comment:
@Krovatkin what does this do? is this going to run a whole bunch of tests on the lazy device if we just wrote 'only_for="lazy"'?

Reply:
yes